sdk/scheduler/src/main/java/com/mesosphere/sdk/scheduler/AbstractScheduler.java

```diff
@@ -213,16 +216,44 @@ public void error(SchedulerDriver driver, String message) {
         SchedulerUtils.hardExit(SchedulerErrorCode.ERROR);
     }
 
+    protected void suppressOrRevive(PlanCoordinator planCoordinator) {
+        synchronized (suppressReviveLock) {
+            if (planCoordinator.hasOperations()) {
+                if (StateStoreUtils.isSuppressed(stateStore)) {
+                    revive();
+                } else {
+                    LOGGER.info("Already revived.");
+                }
+            } else {
+                if (StateStoreUtils.isSuppressed(stateStore)) {
+                    LOGGER.info("Already suppressed.");
+                } else {
+                    suppress();
+                }
+            }
+        }
+    }
+
     void suppress() {
-        LOGGER.info("Suppressing offers.");
-        driver.suppressOffers();
-        StateStoreUtils.setSuppressed(stateStore, true);
+        setOfferMode(true);
     }
 
     void revive() {
-        LOGGER.info("Reviving offers.");
-        driver.reviveOffers();
-        StateStoreUtils.setSuppressed(stateStore, false);
+        setOfferMode(false);
+    }
+
+    private void setOfferMode(boolean suppressed) {
+        synchronized (suppressReviveLock) {
+            if (suppressed) {
+                LOGGER.info("Suppressing offers.");
+                driver.suppressOffers();
```

Comment:
Yes, by design.  If you can't write down your suppress/revive state, the Scheduler has no business staying up.